source/slang/slang-ir-translate-glsl-global-var.cpp

```diff
@@ -223,6 +228,83 @@ namespace Slang
                 entryPointFunc->setFullType(newFuncType);
             }
         }
+
+        // If we see a `GetWorkGroupSize` instruction, we should materialize it by replacing its uses with a constant
+        // that represent the workgroup size of the calling entrypoint.
+        // This is trivial if the `GetWorkGroupSize` instruction is used from a function called by one entry point.
+        // If it is used in a place reachable from multiple entry points, we will introduce a global variable to represent
+        // the workgroup size, and replace the uses with a load from the global variable.
+        // We will assign the value of the global variable at the start of each entry point.
+        //
+        void materializeGetWorkGroupSize(IRModule* module, Dictionary<IRInst*, HashSet<IRFunc*>>& referenceGraph, IRInst* workgroupSizeInst)
+        {
+            IRBuilder builder(workgroupSizeInst);
+            traverseUses(workgroupSizeInst, [&](IRUse* use)
+                {
+                    if (auto parentFunc = getParentFunc(use->getUser()))
+                    {
+                        auto referenceSet = referenceGraph.tryGetValue(parentFunc);
+                        if (!referenceSet) return;
```

Comment:
```suggestion
                        if (!referenceSet) 
                        	return;
```
To allow code stepping